src/backend/distributed/connection/placement_connection.c

```diff
@@ -47,17 +49,14 @@ typedef struct ConnectionReference
 	bool hadDML;
 	bool hadDDL;
 
-	/*
-	 * Membership in ConnectionPlacementHashEntry->connectionReferences or
-	 * ColocatedPlacementsHashEntry->connectionReferences.
-	 */
-	dlist_node placementNode;
```

Comment:
I'm not quite sure why you changed this to achieve the larger goal?  It still seems useful to get other connections that touched a placement?